computer_vision/object_classification/weight_converter.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file object_classification.hpp
+ * @author Kartik Dutt
+ *
+ * Contains implementation of object classification suite. It can be used
+ * to select object classification model, it's parameter dataset and
+ * other training parameters.
+ *
+ * NOTE: This code needs to be adapted as this implementation doesn't support
+ *       Command Line Arguments.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <dataloader/dataloader.hpp>
+#include <models/darknet/darknet.hpp>
+#include <utils/utils.hpp>
+#include <ensmallen_utils/print_metric.hpp>
+#include <ensmallen_utils/periodic_save.hpp>
+#include <ensmallen.hpp>
+#include <mlpack/methods/ann/layer_names.hpp>
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace arma;
+using namespace std;
+using namespace ens;
+
+struct output : public boost::static_visitor<>
+{
+  template <class T>
+  void operator()(T t) const
+  {
+    // We could use this as a summary.
+    LayerNameVisitor a1;
+    std::cout << a1.LayerString(&t) << std::endl;
```

Comment:
I can't simply do t->Parameter() or t.Parameter() here since not all layer have that function but Convolutional layers do. We would have to check if a layer has that function or not first. Interestingly this could be used as model summary function. Currently it outputs "unsupported" layer name.

Reply:
Right, `enable_if` should be helpful here.

Reply:
Agreed, will try it out.